### PR TITLE
More list comprehensions

### PR DESCRIPTION
**Pull request by andlaus**
This replaces `create_sdgs_from_et()` and `create_company_datas_from_et()` by list comprehensions. The reasons are that these functions were pretty trivial, doing it via list comprehensions is more explicit and it is even slightly shorter overall.

Andreas Lauser &lt;andreas.lauser@mercedes-benz.com&gt;, on behalf of [MBition GmbH](https://mbition.io/).
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)

### CODE REVIEW

File: odxtools/comparamsubset.py [hunk 1]
```diff
@@ -45,7 +43,10 @@ def from_et(et_element: ElementTree.Element,
         kwargs = dataclass_fields_asdict(IdentifiableElement.from_et(et_element, doc_frags))
 
         admin_data = AdminData.from_et(et_element.find("ADMIN-DATA"), doc_frags)
-        company_datas = create_company_datas_from_et(et_element.find("COMPANY-DATAS"), doc_frags)
+        company_datas = NamedItemList([
+            CompanyData.from_et(cde, doc_frags)
+            for cde in et_element.iterfind("COMPANY-DATAS/COMPANY-DATA")
```
Comment (kayoub5) [inline odxtools/comparamsubset.py:48]: Maybe there should be a common base class for SDG / ADMIN DATA / COMPANY-DATA ?

Comment (andlaus) [reply odxtools/comparamsubset.py:48]: no: these are all store quite different data for different purposes, cf the XSD...

Comment (kayoub5) [reply odxtools/comparamsubset.py:48]: what I meant is that classes that have company data, should use a common base class, not that sdg and company data should have a common class

Comment (andlaus) [reply odxtools/comparamsubset.py:48]: we should stay with what the XSD defines, lest it become very confusing...

**Review by kayoub5 — APPROVED** [review]

**Pull request merged**